staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -2353,6 +2353,64 @@ func Test_shouldSyncUpdatedNode_compoundedPredicates(t *testing.T) {
 
 }
 
+func Test_shouldSyncUpdatedNode_zoneLabels(t *testing.T) {
```

Comment:
These could be added to existing test table: `Test_shouldSyncUpdatedNode_individualPredicates`. At the same time, that one is growing and growing and that doesn't help readability. I will let you + a maintainer have final say.